include/seqan3/core/algorithm/detail/execution_handler_parallel.hpp

```diff
@@ -189,29 +183,61 @@ class execution_handler_parallel
     {
         assert(state != nullptr);
 
-        if (!state->is_waiting)
+        state->stop_and_wait();
+    }
+
+private:
+    /*!\brief An internal state stored on the heap to allow safe move construction/assignment of the class.
+     *
+     * \details
+     * 
+     * ### Thread safety
+     * 
+     * This class is only intended for use with a single producer model.
```

Comment:
I think we need to put that as a warning comment in the class documentation of the execution handler. Because there will be rather looking there instead of the internal class documentation. 
So the message should convey that this execution handler is only thread safe in a single producer context but allows multiple consumers. And that concurrent invocation of the interfaces will lead to undefined behaviour.

Reply:
```
 * \warning This class is only thread-safe in a single producer context. Multiple consumers are allowed.
 *          Concurrent invocation of the interfaces are undefined behaviour.
```

? That''s what I'll add.